cnxeasybake/tests/test_cli.py

```diff
@@ -7,11 +7,28 @@
 # ###
 """CLI tests."""
 import os
+import sys
+import tempfile
 import unittest
-try:
-    from unittest import mock
-except ImportError:
-    import mock
+
+from contextlib import contextmanager
+from io import StringIO
+
+
+# noqa from http://stackoverflow.com/questions/4219717/how-to-assert-output-with-nosetest-unittest-in-python
+@contextmanager
+def captured_output():
+    if sys.version_info[0] == 3:
+        new_out, new_err = StringIO(), StringIO()
+    else:
+        from io import BytesIO
+        new_out, new_err = BytesIO(), BytesIO()
+    old_out, old_err = sys.stdout, sys.stderr
+    try:
+        sys.stdout, sys.stderr = new_out, new_err
+        yield sys.stdout, sys.stderr
+    finally:
+        sys.stdout, sys.stderr = old_out, old_err
```

Comment:
@pumazi These cli tests are giving me conniptions! Using this contextmanager (that I cribbed from you :-) ) the test run in isolation, as so: `python setup.py test -s cnxeasybake.tests.test_cli.CliTestCase`, but as soon as a logger is initialized, I get a traceback (captured on stderr, though :-) )
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 882, in emit
    stream.write(fs % msg)
TypeError: 'unicode' does not have the buffer interface
Logged from file oven.py, line 186
``` and so on.

Any clever ideas?